uber/models.py

```diff
@@ -994,6 +994,8 @@ class Attendee(MagModel, TakesPaymentMixin):
     group_id = Column(UUID, ForeignKey('group.id', ondelete='SET NULL'), nullable=True)
     group = relationship(Group, backref='attendees', foreign_keys=group_id, cascade='save-update,merge,refresh-expire,expunge')
 
+    promo_code_id = Column(UUID, ForeignKey('promo_code.id'), nullable=True)
```

Comment:
What you want here is more like the group to attendee relationship (one group, many attendees), so you can copy the line above. This should get rid of the cascade deleting as well:
```
promo_code = relationship(PromoCode, backref='used_by', foreign_keys=promo_code_id, cascade='save-update,merge,refresh-expire,expunge')
```